### DISCUSSION TITLE
Is Valtio a good tool for async events?

### DISCUSSION THREAD

**Discussion by niamleeson** [pmndrs/valtio#904] [Q&A]
I understand Valtio is great for rapid vanilla js state updates outside of the known UI libraries. What I couldn't easily grasp from reading the docs is whether Valtio can be useful for combining existing state with user events like input change, mouse and hover/movement, and then trigger api calls in certain scenarios. Or do use cases like this rather require observables like RxJs? Would you kindly help me understand more complex use cases of Valtio?

**Comment by dai-shi** [comment]
Valtio proxies are very lightweight nested proxies. Theoretically, you can use them for any values such as events.
But, there's a caveat. Its react binding is "async" by default.
So, using valtio value for `<input>` doesn't work unless you opt-in to "sync". https://github.com/pmndrs/valtio#update-synchronously
So, that may imply Valtio isn't designed for user events.
But, as far as I understand observerbles are async too, and it might not help in this case.
